profiling/roctx-connector/kp_roctx_connector.cpp

```diff
@@ -61,66 +74,120 @@ KokkosP: Finalization of ROC Tracer Connector. Complete.
-                                            uint64_t* /*kID*/) {
+void kokkosp_begin_parallel_scan(const char* name,
+                                 const uint32_t /*devID*/,
+                                 uint64_t* /*kID*/) {
   roctxRangePush(name);
 }
 
-extern "C" void kokkosp_end_parallel_scan(const uint64_t /*kID*/) {
+void kokkosp_end_parallel_scan(const uint64_t /*kID*/) {
   roctxRangePop();
 }
 
-extern "C" void kokkosp_begin_parallel_reduce(const char* name,
-                                              const uint32_t /*devID*/,
-                                              uint64_t* /*kID*/) {
+void kokkosp_begin_parallel_reduce(const char* name,
+                                   const uint32_t /*devID*/,
+                                   uint64_t* /*kID*/) {
   roctxRangePush(name);
 }
 
-extern "C" void kokkosp_end_parallel_reduce(const uint64_t /*kID*/) {
+void kokkosp_end_parallel_reduce(const uint64_t /*kID*/) {
   roctxRangePop();
 }
 
-extern "C" void kokkosp_push_profile_region(char* name) {
+void kokkosp_push_profile_region(const char* name) {
   roctxRangePush(name);
 }
 
-extern "C" void kokkosp_pop_profile_region() { roctxRangePop(); }
+void kokkosp_pop_profile_region() { roctxRangePop(); }
 
-extern "C" void kokkosp_create_profile_section(const char* name,
-                                               uint32_t* sID) {
+void kokkosp_create_profile_section(const char* name,
+                                    uint32_t* sID) {
   *sID = kokkosp_sections.size();
   kokkosp_sections.push_back(
       {std::string(name), static_cast<roctx_range_id_t>(-1)});
 }
 
-extern "C" void kokkosp_start_profile_section(const uint32_t sID) {
+void kokkosp_start_profile_section(const uint32_t sID) {
   auto& section = kokkosp_sections[sID];
   section.id    = roctxRangeStart(section.label.c_str());
 }
 
-extern "C" void kokkosp_stop_profile_section(const uint32_t sID) {
+void kokkosp_stop_profile_section(const uint32_t sID) {
   auto const& section = kokkosp_sections[sID];
   roctxRangeStop(section.id);
 }
 
-extern "C" void kokkosp_destroy_profile_section(const uint32_t sID) {
+void kokkosp_destroy_profile_section(const uint32_t sID) {
   // do nothing
 }
 
-extern "C" void kokkosp_begin_fence(const char* name, const uint32_t /*devID*/,
-                                    uint64_t* fID) {
+void kokkosp_profile_event(const char* name) { roctxMark(name); }
+
+void kokkosp_begin_fence(const char* name, const uint32_t /*devID*/,
+                         uint64_t* fID) {
   *fID = roctxRangeStart(name);
 }
 
-extern "C" void kokkosp_end_fence(const uint64_t fID) { roctxRangeStop(fID); }
+void kokkosp_end_fence(const uint64_t fID) { roctxRangeStop(fID); }
+
+Kokkos::Tools::Experimental::EventSet get_event_set() {
+  Kokkos::Tools::Experimental::EventSet my_event_set;
+  memset(&my_event_set, 0,
+         sizeof(my_event_set));  // zero any pointers not set here
```

Comment:
Uggg.  That thing really something that should be handled at construction.
I suspect value-initialization
```C++
EventSet my_event_set{};
```
does what you want.

Reply:
Indeed :) I used this piece of code because all connectors are currently doing it this way. It may be best to address the issue for all connectors in a separate PR? One way forward might be to provide default initialisers using `= NULL` for the function pointers in the declaration of the struct in `Kokkos_Profiling_C_interface.h`?

Reply:
> I used this piece of code because all connectors are currently doing it this way.

You are right.  This is fine then.

And yes definitely fix elsewhere.

Reply:
OK. I made issue 

- https://github.com/kokkos/kokkos-tools/issues/220